packages/components/badge/src/badge.vue

```diff
@@ -41,6 +43,16 @@ const content = computed<string>(() => {
   }
   return `${props.value}`
 })
+const style = computed<StyleValue>(() => {
+  return [
+    props.dotStyle ?? {},
+    {
+      backgroundColor: props.color,
```

Comment:
`backgroundColor` may need to add a judgment. If the background color is set through `dotStyle` without setting `color`, it may not take effect.
```vue
<el-badge :value="1" :dot-style="{background: '#6cf'}">
  <el-button> offset</el-button>
</el-badge>
```